packages/core/src/StateNode.ts

```diff
@@ -976,13 +972,9 @@ class StateNode<
     };
   }
 
-  private nodesFromChild(
+  private getExternalReentryNodes(
     childStateNode: StateNode<TContext, any, TEvent, any, any, any>
   ): Array<StateNode<TContext, any, TEvent, any, any, any>> {
-    if (childStateNode.escapes(this)) {
-      return [];
-    }
```

Comment:
It was only used here, I've replaced this check with the conditional return [here](https://github.com/statelyai/xstate/pull/3180/files#diff-7f553f4d4c285f1c05cced2e979c5b957d8884e39e74cc6797b32c63a0be7e7aR987-R990). Thanks to that we only need 1 tree traversal instead of 2 to compute the result of `getExternalReentryNodes`